docs/DeveloperGuide.md

```diff
@@ -1,33 +1,297 @@
-# Developer Guide
+---
+layout: page
+title: Developer Guide
+---
+
+- Table of Contents
+{:toc}
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
+
+This section provides a high-level explanation of the design and implementation of AthletiCLI, 
+supported by UML diagrams and short code snippets to illustrate the flow of data and interactions between the 
+components.
+
+
+### Architecture
+
+Given below is a quick overview of main components and how they interact with each other.
+<p  align="center" width="100%">
+  <img width="80%" src="images/architectureDiagram.svg" alt="'set-diet-goal' Sequence Diagram"/>
+</p>
+
+**Main components of the architecture**
+
+**`AthletiCLI`** is in charge of the app launch and shut down.
+
+The bulk of the AthletiCLI’s work is done by the following components, with each of them corresponds to a package:
+
+* [`UI`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/ui): The UI and other UI-related sub-components of AthletiCLI.
+* [`Parser`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/parser): Parses the commands input by the users.
+* [`Storage`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/storage): Reads data from, and writes data to, the hard disk.
+* [`Data`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/data): Holds the data of AthletiCLI in memory.
+* [`Commands`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/commands): The command executors.
+
+[`Exceptions`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/exceptions) represents exceptions used by multiple other components.
+
+### Overview
+
+The class diagram shows the relationship between `AthletiCLI`, `Ui`, `Parser`, and `Data`.
```

Comment:
Should your class diagram show the accessibility of the functions in it?
<img width="697" alt="Screenshot 2023-11-02 at 5 23 24 PM" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/65208386/9efcec52-23b2-40f9-b17e-d06e9588eb05">
<img width="677" alt="Screenshot 2023-11-02 at 5 23 32 PM" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/65208386/f1f78f9a-b739-4b2f-b5e8-2ab527eef2a8">

Reply:
Thanks for the feedback! However we cannot seem to find a way to label it with PUML. Perhaps you could share the software that you are using for your class diagrams.

Reply:
Your UML Diagramm looks good to me but it might be better to delete the square used for attributes and methods if this class doesn’t have it at all.